### ISSUE TITLE
Scatter Plot SmoothTension Property takes no effect

### ISSUE THREAD

**Issue by bjschwarz** [ScottPlot/ScottPlot#3274]
**Issue:** (Describe the bug here)
The Smooth Tension Property on the scatter plot does not change the curve of the corresponding plot. 

**ScottPlot Version:** (What NuGet package are you using?)
5.0.20

**Code Sample:** 

```cs
Plot plt = new();

// use code to express information related to the the discussion
DateTime[] days = Generate.DateTime.Days(7);
int[] data = Generate.RandomIntegers(7, 5);

plt.Axes.DateTimeTicks(ScottPlot.Edge.Bottom);

var scatter = plt.Add.Scatter(days, data);
scatter.Smooth = true;
scatter.SmoothTension = 1.0f;
scatter.LineStyle.Width = 3;
scatter.LineStyle.Pattern = LinePattern.Solid;
scatter.MarkerSize = 0;

plt.SavePng("curve.png", 300, 300);

var scatter2 = plt.Add.Scatter(days, data);
scatter2.Smooth = true;
scatter2.SmoothTension = 0.0f;
scatter2.LineStyle.Width = 3;
scatter2.LineStyle.Pattern = LinePattern.Solid;
scatter2.MarkerSize = 0;

plt.SavePng("nocurve.png", 300, 300);
```
![curve](https://github.com/ScottPlot/ScottPlot/assets/14096147/91508899-65c3-46c4-95f8-3e629f92bc08)
![nocurve](https://github.com/ScottPlot/ScottPlot/assets/14096147/8ea4d9d8-fae7-429a-a0e2-1d76410ab3e5)

**Comment by keleutos2022** [comment]
Is there any news on this?

**Comment by swharden** [comment]
Following-up, I think this can be implemented by replacing this line:

https://github.com/ScottPlot/ScottPlot/blob/ad66085d2086fbe3c744f5cb43cd04620517830d/src/ScottPlot5/ScottPlot5/Plottables/Scatter.cs#L66

...with something that detects if `Smooth` is enabled and instead draws using cubic spline interpolation: https://learn.microsoft.com/en-us/dotnet/api/skiasharp.skpath.cubicto?view=skiasharp-2.88

There may be better strategies for drawing lines with adjustable tension. I'm personally focusing on higher priority issues right now, but I definitely want to support this eventually and welcome suggestions from the community along the way 🚀

**Comment by swharden** [comment]
I started working on this now and, wow, it's not as easy as I thought. I think the solution will be to use SkiaSharp's conic Bézier curve feature, and add some logic to place the control point in a place that allows for smooth lines.

https://learn.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/graphics/skiasharp/curves/beziers#the-conic-b%C3%A9zier-curve

**Comment by swharden** [comment]
A related strategy is described here https://code.allaboutapps.at/articles/smooth-graph-lines/

> In order for the curve to go through every point, we have to divide each segment into two curves and calculate the control point for each part.

**Comment by swharden** [comment]
Hi everyone, thanks for your input on this! It is clear that there are many different strategies to achieve a "smooth line". I implemented a very basic one in #3606 and plan to merge it in today, satisfying the basic requirement for smooth line support in scatter plots so I'll close this issue, but recognize that additional work could be done to improve this feature.

I think that merging #3606 will make it easier to add additional support for new types of smoothing in the future. If others are interested in extending this work to add more advanced forms of smooth lines, I welcome input or pull requests!

![image](https://github.com/ScottPlot/ScottPlot/assets/4165489/ec853963-f1d3-4a0b-93ec-7ed38b4fb5e4)
